runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -10338,6 +10338,307 @@ inlineCompareAndSwapNative(
+   generateRegRegInstruction(TR::InstOpCode::PCMPGTBRegReg, node, xmmMaskReg, xmmChunkReg, cg);
+
+   // Extract bitmask of negative values
+   generateRegRegInstruction(TR::InstOpCode::PMOVMSKB4RegReg, node, maskReg, xmmMaskReg, cg);
+
+   // Check if any negative values exist
+   generateRegRegInstruction(TR::InstOpCode::TEST4RegReg, node, maskReg, maskReg, cg);
+
+   // If the result is nonzero, handle finding a negative byte
+   if (recognizedMethod == TR::java_lang_StringCoding_hasNegatives)
+      {
+      // Jump to returnHasNegativesLabel
+      generateLabelInstruction(TR::InstOpCode::JNE4, node, returnHasNegativesLabel, cg);
+      }
+   else
+      {
+      // Break and calculate index of first negative byte
+      generateLabelInstruction(TR::InstOpCode::JNE1, node, foundNegativeLabel, cg);
+      }
+
+   // index += 16
+   generateRegImmInstruction(TR::InstOpCode::ADD4RegImm4, node, indexReg, 16, cg);
+
+   // Jump back to loopLabel
+   generateLabelInstruction(TR::InstOpCode::JMP1, node, loopLabel, cg);
+
+
+   // If we found any negative bytes, calculate the index of the first negative byte in the chunk
+   if (recognizedMethod == TR::java_lang_StringCoding_countPositives)
+      {
+      generateLabelInstruction(TR::InstOpCode::label, node, foundNegativeLabel, cg);
+      generateRegRegInstruction(TR::InstOpCode::TZCNT4RegReg, node, maskReg, maskReg, cg);
+      generateRegRegInstruction(TR::InstOpCode::ADD4RegReg, node, indexReg, maskReg, cg);
+
+      // Jump to returnHasNegativesLabel
+      generateLabelInstruction(TR::InstOpCode::JMP4, node, returnHasNegativesLabel, cg);
+      }
+
+
+   // Deal with the residual (last 15 or fewer) bytes
+   generateLabelInstruction(TR::InstOpCode::label, node, residualLabel, cg);
+
+   // Calculate bytes remaining: loopLimit = -(loopLimit - limit)
+   generateRegRegInstruction(TR::InstOpCode::SUB4RegReg, node, loopLimitReg, limitReg, cg);
+   generateRegInstruction(TR::InstOpCode::NEG4Reg, node, loopLimitReg, cg);
+
+   /*
+    *    if loopLimit == 0
+    *       jmp returnNoNegativesLabel
+    *    if loopLimit > 8
+    *       jmp nineOrMoreBytesLabel ----+
+    *    if loopLimit > 2                |
+    *       jmp threeOrMoreBytesLabel -+ |
+    *                                  | |
+    *    load 1-2 bytes                | |
+    *    jmp residualTestLabel         | |
+    *                                  | |
+    *    threeOrMoreBytesLabel: <------+ |
+    *       if loopLimit > 4             |
+    *       jmp fiveOrMoreBytesLabel --+ |
+    *                                  | |
+    *       load 3-4 bytes             | |
+    *       jmp residualTestLabel      | |
+    *                                  | |
+    *    fiveOrMoreBytesLabel: <-------+ |
+    *       load 5-8 Bytes               |
+    *       jmp residualTestLabel        |
+    *                                    |
+    *    nineOrMoreBytesLabel: <---------+
+    *       load 9-16 bytes
```

Comment:
You should round out the pseudo-code with the rest of the residue handing, otherwise it is incomplete.